### ISSUE TITLE
Integer overflow in auc

### ISSUE THREAD

**Issue by mvanhala** [mlr-org/mlr3measures#8]
In [auc](https://github.com/mlr-org/mlr3measures/blob/master/R/binary_auc.R), if `n_pos * (n_pos + 1L)` or `n_pos * n_neg` exceed the maximum integer size, the result will be `NA` due to integer overflow.

```
set.seed(123)
truth <- factor(sample(c("Y", "N"), 250000, replace = TRUE))
prob <- runif(250000)
mlr3measures::auc(truth, prob, "Y")
#> Warning in n_pos * (n_pos + 1L): NAs produced by integer overflow
#> Warning in n_pos * n_neg: NAs produced by integer overflow
#> [1] NA
```

**Comment by mvanhala** [comment]
I've created pull request #9 for this issue.

**Comment by mllg** [comment]
Thanks for reporting and thanks for the PR. Fixed via #9.
